Completion/Unix/Command/_code

```diff
@@ -0,0 +1,36 @@
+#compdef code
+
+local arguments
+
+arguments=(
+  '(-d --diff)'{-d,--diff}'[compare two files with each other]:file to compare:_files:file to compare with:_files'
+  '(-a --add)'\*{-a,--add}'[add specified directory (or directories) to the last active window.]:directory:_directories'
```

Comment:
You don't want the (-a --add) exclusion as well as \*.
The * prevents -a excluding -a (i.e. itself) which it does by default. The option can be repeated so you can mix -a and --add. So no exclusion is needed.

I wouldn't bother with the "(or directories)". The fact that the option can be repeated conveys the information that more than one directory can be added. Admittedly, _arguments doesn't make that as obvious as it might in completion listings but it'd be better to try to make it do so for all completions than to use convoluted wording in each one.

Reply:
Agreed. Removed `(-a --add)` exclusion and redundant pluralization.